app/controllers/users/registrations_controller.rb

```diff
@@ -21,6 +21,12 @@ def new
 
   # POST /resource
   def create
+    # We may need the confirmation mailer to access the current procedure.
+    # But there's no easy way to pass an argument to the mailer through
+    # all Devise code.
+    # So instead we use a per-request global variable.
+    CurrentConfirmation.procedure_after_confirmation = @procedure
+
```

Comment:
@LeSim here is the contentieux :)

Dans ce chemin de code Devise peut envoyer un email de confirmation de deux manières :

- soit dans `existing_user.resend_confirmation_instructions`
- soit dans `super`
  - qui appelle `User.send_confirmation_instructions`
    - qui find_or_create, puis appelle `user.send_confirmation_instructions`
      - qui appelle `DeviseUserMailer.confirmation_instructions`

Idéalement on devrait passer l'ID de procédure par argument. Mais ça nécessite de percer un trou à travers toute cette chaine d'appels intermédiaires de Devise. Ce qui veut dire tout overrider, copier-coller, etc.

Sinon, dans les solutions alternatives :

- Y'a pas vraiment moyen de stocker la donnée sur une instance de `user`, parce que c'est Devise qui `find_or_create` l'instance lui-même, avec sa propre logique intercalée ;
- Y'a pas vraiment moyen d'appeler le mailer directement nous même, parce que Devise a de la logique de "Est-ce que c'est une confirmation ou une re-confirmation ?" qu'on veut conserver.
- Y'a pas vraiment moyen de stocker ça dans la session, parce que le mailer n'a pas accès aux sessions (et en plus c'est jamais qu'un autre genre de variable globale).

Au final je suis donc parti sur ce genre de variables globale. C'est pas super élégant. Mais à défaut d'autre chose, ça a une portée très limitée ; `procedure_after_confirmation` est facilement greppable dans le code, et n'est utilisé que pour ça. Et c'est nettoyé automatiquement à la fin de la requête.

En tout cas si tu vois une alternative que tu préfèrerais, on peut en parler.

Reply:
@tchak @Keirua @krichtof maintenant que Sim est en vacances, je veux bien votre œil aussi :)

Reply:
@kemenaran Instinctivement, j'aurai tenté aussi la 1ere alternative, à savoir stocker la donné sur l'instance de `user`.

Mais si je comprends bien, actuellement (avant ta PR), l'info est stockée dans la session. Et maintenant, tu proposes de le stocker au niveau de la requête, donc une portée moindre. On y gagne en terme d'élégance :)

Reply:
Ca me va aussi, c'est pas un gros hack et c'est même plutôt élégant comparé aux alternatives